accord-core/src/main/java/accord/primitives/Unseekables.java

```diff
@@ -45,7 +49,7 @@ public boolean isFullRoute()
                     return right;
 
                 // non-route types can always accept route types as input, so just call its union method on the other
-                return leftKind.isRoute() ? ((Unseekables)right).union(left) : ((Unseekables)left).union(right);
+                return left.with(right);
```

Comment:
Not quite, I will document them more clearly. Though the distinction is only relevant for `Route`, for which `union` will return another `Route` whereas `with` will return an `Unseekables`. `union` definitely contains more information than either parameter, whereas `with` may downgrade from a `Route` in order to safely expand the `Unseekable` that are contained in either.

Reply:
It might be that we can actually get rid of `merge` now and just use `with`. I am open to better names.

We could also get rid of `union` from `Routables`, `Unseekables` and `Seekables`, leaving it only for `Route`, and have only `with` in the other three. I will prepare this change and see how we like it.

Reply:
Ok, so I have kept all three, but modified the hierarchies and documented the differences. The idea behind `merge` is that it uses the strongest merge possible, so `union` if possible and `with` otherwise, as now defined.

Reply:
`UnionMode.mergeAdjacent` doesn't seem to be used.

At the moment it makes sense. There are a lot of methods for merging these things, but also a lot of them have little bits of specialization for doing things like avoiding constructing a new instance.

Reply:
I would prefer to keep the facility there, as we may want to use it in future to minimise `Ranges`. I assume I did use it before.

I usually stick to the rule of "not using it, delete it" but with code still in flux I prefer to avoid repeating work in the near future. Some of these facilities can be cleaned up before we release if they remain unused.